language/move-vm/transactional-tests/tests/instructions/arithmetic_operators_u16.mvir

```diff
@@ -0,0 +1,164 @@
+//# run
+main() {
```

Comment:
Can you split these tests into positive and negative tests? It is a bit confusing to have some operations that should fail and some that shouldn't in the same test, makes it hard to see what exactly is getting coverage.

Reply:
@tnowacki I copied the same format which existed for u8, u64, u128, which also mixed the passes and fails.
I 100% agree it should be separated (was hard for me to read too) but this faulty pattern is repeated all over the codebase.
If I were fixing tests, this PR would be much much bigger so now I wanna focus on the core essence of adding the requested integer support.
We can file a cleanup PR as needed.